### PR TITLE
[DOCS] Adds higher-level monitoring configuration page

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds a "Configuring monitoring" landing page in the Elasticsearch Reference, such that you don't land immediately in the older monitoring instructions. It matches the layout in Kibana (https://www.elastic.co/guide/en/kibana/master/configuring-monitoring.html).  The new table of contents is as follows:

image

NOTE: The "new" collecting-monitoring-data.asciidoc is just a renamed copy of the existing configuring-monitoring.asciidoc.  The only change is the page level and the addition of a NOTE in the intro.

### CODE REVIEW

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+node or across the cluster as a dynamic cluster setting. If {es} 
+{security-features} are enabled, you must have `monitor` cluster privileges to 
+view the cluster settings and `manage` cluster privileges to change them.
+
+For more information, see <<monitoring-settings>> and <<cluster-update-settings>>.
+--
+
+.. Set the `xpack.monitoring.collection.enabled` setting to `true` on each
+node in the cluster. By default, it is is disabled (`false`). 
++
```
Comment (ycombinator) [inline docs/reference/monitoring/collecting-monitoring-data.asciidoc]: This setting can also be toggled from the Monitoring UI itself by clicking the "Turn on Monitoring" button. We allude to this in https://www.elastic.co/guide/en/kibana/master/monitoring-kibana.html as well ("To update the cluster settings in Kibana..."). Should we mention the UI method here as well?

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      truststore.path: /path/to/file
+      truststore.password: password
+--------------------------------------------------
+--
+
+. Configure your cluster to route monitoring data from sources such as {kib}, 
+Beats, and {ls} to the monitoring cluster. The 
+`xpack.monitoring.collection.enabled` setting must be `true` on each node in the 
+cluster. For information about configuring each product to collect and send 
+monitoring data, see {stack-ov}/xpack-monitoring.html[Monitoring the {stack}].
```
Comment (ycombinator) [inline docs/reference/monitoring/collecting-monitoring-data.asciidoc]: Isn't this information mentioned earlier already?

> .. Set the `xpack.monitoring.collection.enabled` setting to `true` on each
node in the cluster. By default, it is is disabled (`false`).

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
++
+--
+TIP: You may want to temporarily {ref}/modules-cluster.html[disable shard
+allocation] before you restart your nodes to avoid unnecessary shard
+reallocation during the install process.
+
+--
+
+. Optional: 
+<<config-monitoring-indices,Configure the indices that store the monitoring data>>.
```
Comment (ycombinator) [inline docs/reference/monitoring/collecting-monitoring-data.asciidoc]: Isn't this information mentioned earlier already?

> .. Optional: Specify which indices you want to monitor.

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/monitoring/collecting-monitoring-data.asciidoc]: Done!

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/reference/monitoring/collecting-monitoring-data.asciidoc]: You're right.  Not sure why that was repeated.

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc (hunk 3 above)
Comment (lcawl) [reply docs/reference/monitoring/collecting-monitoring-data.asciidoc]: By my understanding, those steps are different.  One is specifying which indices to monitor, the other is tweaking the .monitoring-* indices. If that's not correct or not clear, let me know!

**Comment by lcawl** [comment]
Thanks for the feedback @ycombinator . In case it helps, I've thrown up a preview of the new layout here: https://doc-review-63bf6.firebaseapp.com/es3/configuring-monitoring.html

File: docs/reference/monitoring/collecting-monitoring-data.asciidoc (hunk 3 above)
Comment (ycombinator) [reply docs/reference/monitoring/collecting-monitoring-data.asciidoc]: You're right, I didn't read properly.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Comment by lcawl** [comment]
Verified that `gradlew -p docs check` completes successfully.

**Pull request merged**